src/librbd/deep_copy/ObjectCopyRequest.cc

```diff
@@ -144,7 +180,8 @@ void ObjectCopyRequest<I>::send_read_object() {
       return;
     }
 
-    send_write_object();
+    compute_parent_object_extents();
```

Comment:
You should only need to read from the parent on the first snapshot (or HEAD revision) that you are copying. Should the ```read_from_parent``` logic be moved to ```compute_read_ops``` for clarity? If it's ```m_snap_set``` does not contain any extents for ```start_src_snap_id == 0``` and the object overlaps w/ the parent for ```end_src_snap_id```, you will have to read from the parent and you can execute that before running ```send_read_object```.

When you do read from the parent, if the data is all zero, you will have to add a write-op for ```create```, otherwise it's just a write-op for the data read from the parent. In the future, we can improve our internal IO read path to keep IO sparse, but for now I'd say don't worry about trying to keep the destination sparse if some parent data exists.